civis/tests/test_futures.py

```diff
@@ -183,6 +183,20 @@ def test_set_api_result_failed(self, mock_subscribe, mock_api):
         assert result._state == 'FINISHED'
         with pytest.raises(CivisJobFailure):
             result.result()
+        with pytest.raises(CivisJobFailure):
+            result.outputs()
+
+    def test_outputs_succeeded(self):
```

Comment:
Can we add a test for if the job fails? Does `list_run_outputs` return a sensible result?

Reply:
If the job fails, it will fail during the `.result()` call, and the user will see the same error that they would get from `.result()`. My hope is that this would be an error a user would be familiar with. The addition to the test on line 186 there is meant to check for this case.